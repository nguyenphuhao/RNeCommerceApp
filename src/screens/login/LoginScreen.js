import React, { useState } from 'react';
import { View, Platform, KeyboardAvoidingView, Alert } from 'react-native';
import { Input, Text, Button } from 'react-native-elements';
import Icon from 'react-native-vector-icons/FontAwesome';
import LoginBackground from '../../assets/img/login-background.svg';
import styles from './styles';

import { login } from '../../services/login';

const LoginScreen = () => {
  const [emailAddress, setEmailAddress] = useState('');
  const [password, setPassword] = useState('');

  const authenticate = () => {
    login({ loginname: emailAddress, password: password })
      .then(token => {
        Alert.alert('Login', 'Login successfully.', [
          { text: 'OK', onPress: () => console.log('OK Pressed') },
        ]);
      })
      .catch(error => {
        Alert.alert('Login', error.message, [
          { text: 'OK', onPress: () => console.log('OK Pressed') },
        ]);
      });
  };

  return (
    <>
      <LoginBackground
        width={'100%'}
        height={'100%'}
        style={styles.background}
      />
      <View style={styles.container}>
        <Text testId={'tstCaption'} style={styles.caption} h3>
          Ecommerce{'\n'}Store
        </Text>
        <KeyboardAvoidingView
          style={styles.form}
          behavior={Platform.Os === 'ios' ? 'padding' : 'height'}
        >
          <Input
            testId={'tstEmailAddress'}
            placeholder={'Email Address'}
            value={emailAddress}
            onChangeText={setEmailAddress}
          />
          <Input
            testId={'tstPassword'}
            placeholder={'Password'}
            secureTextEntry={true}
            value={password}
            onChangeText={setPassword}
          />
          <Text testId={'tstForgotPassword'} style={styles.forgotPasswordText}>
            Forgot Password ?
          </Text>
        </KeyboardAvoidingView>
        <View style={styles.action}>
          <Button
            testId={'tstSignIn'}
            title="SIGN IN"
            onPress={() => {
              authenticate();
            }}
          />
          <Text testId={'tstSignUp'} style={styles.signUpText}>
            New Here? Sign Up
          </Text>
        </View>
        <Button
          testId={'tstSkipLogin'}
          containerStyle={styles.skipLogin}
          type="clear"
          icon={<Icon name="arrow-right" size={20} color="#00a8f3" />}
          title="SKIP LOGIN "
          iconRight
        />
      </View>
    </>
  );
};

export default LoginScreen;
