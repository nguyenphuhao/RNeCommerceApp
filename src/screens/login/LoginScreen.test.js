/**
 * @format
 */

import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import LoginScreen from './LoginScreen';

const shallowComponent = () => {
  const props = {};
  const wrapper = shallow(<LoginScreen {...props} />);
  return wrapper;
};

describe('LoginScreen exists components', () => {
  test('LoginScreen exists components - test snapshot', () => {
    const wrapper = shallowComponent();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
