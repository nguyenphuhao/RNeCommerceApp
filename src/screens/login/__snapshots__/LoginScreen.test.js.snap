// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen exists components LoginScreen exists components - test snapshot 1`] = `
<Fragment>
  <SvgMock
    height="100%"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    width="100%"
  />
  <View
    style={
      Object {
        "flex": 12,
        "padding": 10,
      }
    }
  >
    <Themed.Text
      h3={true}
      style={
        Object {
          "color": "#00a8f3",
          "elevation": 1,
          "flex": 6,
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
      testId="tstCaption"
    >
      Ecommerce
      

      Store
    </Themed.Text>
    <KeyboardAvoidingView
      behavior="height"
      enabled={true}
      keyboardVerticalOffset={0}
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <ForwardRef(Themed.Input)
        onChangeText={[Function]}
        placeholder="Email Address"
        testId="tstEmailAddress"
        value=""
      />
      <ForwardRef(Themed.Input)
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        testId="tstPassword"
        value=""
      />
      <Themed.Text
        style={
          Object {
            "color": "#436bff",
            "fontSize": 18,
            "textAlign": "right",
          }
        }
        testId="tstForgotPassword"
      >
        Forgot Password ?
      </Themed.Text>
    </KeyboardAvoidingView>
    <View
      style={
        Object {
          "flex": 2,
          "marginTop": -20,
        }
      }
    >
      <ForwardRef(Themed.Button)
        onPress={[Function]}
        testId="tstSignIn"
        title="SIGN IN"
      />
      <Themed.Text
        style={
          Object {
            "color": "#436bff",
            "fontSize": 18,
            "marginTop": 6,
            "textAlign": "center",
          }
        }
        testId="tstSignUp"
      >
        New Here? Sign Up
      </Themed.Text>
    </View>
    <ForwardRef(Themed.Button)
      containerStyle={
        Object {
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
      icon={
        <Icon
          allowFontScaling={false}
          color="#00a8f3"
          name="arrow-right"
          size={20}
        />
      }
      iconRight={true}
      testId="tstSkipLogin"
      title="SKIP LOGIN "
      type="clear"
    />
  </View>
</Fragment>
`;
